integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.component.spec.ts

```diff
@@ -3,4 +3,86 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-// TODO: Add unit tests
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  ElementRef,
+} from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { OpfGooglePayComponent } from './google-pay.component';
+import { OpfGooglePayService } from './google-pay.service';
+
+class MockOpfGooglePayService {
+  loadProviderResources = jasmine
+    .createSpy()
+    .and.returnValue(Promise.resolve());
+  initClient = jasmine.createSpy();
+  isReadyToPay = jasmine
+    .createSpy()
+    .and.returnValue(Promise.resolve({ result: true }));
+  renderPaymentButton = jasmine.createSpy();
+}
+
+describe('OpfGooglePayComponent', () => {
+  let component: OpfGooglePayComponent;
+  let fixture: ComponentFixture<OpfGooglePayComponent>;
+  let mockOpfGooglePayService: MockOpfGooglePayService;
+  let mockChangeDetectorRef: jasmine.SpyObj<ChangeDetectorRef>;
+
+  beforeEach(async () => {
+    mockChangeDetectorRef = jasmine.createSpyObj('ChangeDetectorRef', [
+      'detectChanges',
+    ]);
+    mockOpfGooglePayService = new MockOpfGooglePayService();
+
+    TestBed.configureTestingModule({
+      declarations: [OpfGooglePayComponent],
+      providers: [
+        { provide: OpfGooglePayService, useValue: mockOpfGooglePayService },
+        { provide: ChangeDetectorRef, useValue: mockChangeDetectorRef },
+      ],
+    })
+      .overrideComponent(OpfGooglePayComponent, {
+        set: { changeDetection: ChangeDetectionStrategy.OnPush },
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(OpfGooglePayComponent);
+    component = fixture.componentInstance;
+    component.activeConfiguration = {};
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should initialize Google Pay client on init', async () => {
+    fixture.detectChanges();
+    await fixture.whenStable();
```

Comment:
We are duplicating these two lines in several blocks. What do you think about creating a function with that and just call a function?

Reply:
Changed